### DISCUSSION TITLE
MachineConfig housekeeping

### DISCUSSION THREAD

**Discussion by peterroth** [okd-project/okd#1814] [General]
Hello all,
I realized we have a lot of machine configs on our clusters, one of them carries 145 (16 infra, 64 master and 65 worker). This cluster was updated from OKD 4.8 to 4.13 through the years, and I found rendered configs even from 4.10.0-0.okd-2022-06-10-131327.
If my understanding is correct, only the last is used, meaning the rest isn't really necessary (we can surely keep the last 2-3 for security reasons).
Hence my question is: how you deal with the machine configs, how to safely remove old versions? Shouldn't the MCO do it?

**Comment by ibotty** [comment]
When debugging machineconfig issues it's always nice to not have to filter that many `mc`s.

It's documented to leave them alone, but I figure it would be great if the operator were to remove old ones.

**Reply by peterroth** [reply]
I'm 100% sure I won't read a machine config from 2022 March :)
Maybe a period would be useful, for example delete ones that are older than half a year, but keep the last X.
